### DISCUSSION TITLE
Jetstream: Publish

### DISCUSSION THREAD

**Discussion by robertmircea** [nats-io/nats.net.v1#607] [Q&A] [answered]
1. Is it possible to publish messages in batch for Jetstream with current nats.net client?
2. Can I publish a fire&forget messages with current jetstream context Publish overloads? For example, in my app I publish audit events to a stream, but I would not like to wait for server confirmation.
3. What is the recommended way to check for publish result: should I check PublishAck's HasError only or should I also catch any NATSException which may result?

**Comment by scottf** [comment, marked as answer]
1.  There is no batch. You could use an async publish, collect the tasks and inspect them later, say every N messages. 
2.A You could just publish async and ignore the task/ack. 
2.B Once you have setup the stream and subject, you can just do a core publish to the subject. Under the covers the Jetstream publish is just a core request/reply where the reply is from the server. There is an example of that here: https://github.com/nats-io/nats.net/blob/master/src/Samples/JetStreamPublishVsCorePublish/JetStreamPublishVsCorePublish.cs
3. I would check that the task finishes without exception (likely a timeout) and the HasError flag.
